samtranslator/model/sam_resources.py

```diff
@@ -2973,11 +2974,13 @@ def _construct_appsync_resolver_resources(
                 # to a default snippet which has basic definition of request/response functions.
                 if not cfn_resolver.Code and not cfn_resolver.CodeS3Location:
                     cfn_resolver.Code = APPSYNC_PIPELINE_RESOLVER_JS_CODE
+                    cfn_resolver.Runtime = APPSYNC_PIPELINE_RESOLVER_JS_RUNTIME
+                else:
+                    cfn_resolver.Runtime = self._parse_runtime(resolver, relative_id)
```

Comment:
Im guessing if code or s3location is set then runtime needs to be set? Do we have an error thats thrown if runtime isn't set?

Reply:
Yes in `_parse_runtime()`